geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java

```diff
@@ -968,8 +968,11 @@ public Socket connect(InetAddress inetadd, int port, int timeout,
 
   /**
    * Will be a server socket... this one simply registers the listeners.
+   *
+   * @param timeout the socket's timeout will be set to this (in milliseconds).
    */
-  public void configureServerSSLSocket(Socket socket) throws IOException {
+  public void configureServerSSLSocket(Socket socket, int timeout) throws IOException {
+    socket.setSoTimeout(timeout);
     if (socket instanceof SSLSocket) {
```

Comment:
This seems like the wrong failure behavior (silently continue) if we call this on a non-SSL socket.

Reply:
It seems like it's mostly the name of this method that is odd. It looks like the server socket gets configured for SSL when it's created, and child sockets will be SSL or not based on what the server socket is.

Perhaps it should be `handshakeIfSocketIsSSL` or something?